sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1829,3 +1858,21 @@ class FakeV2Provider extends TableProvider {
     throw new UnsupportedOperationException("Unnecessary for DDL tests")
   }
 }
+
+class VacuumableTableProvider extends TableProvider {
+
+  override def getTable (options: CaseInsensitiveStringMap): Table =
+    new VacuumableTable
+  class VacuumableTable extends Table with SupportsVacuum {
+
+    override def name(): String = "vacuum"
+
+    override def schema(): StructType =
+      StructType(Seq(StructField("id", IntegerType)))
+
+    override def capabilities(): util.Set[TableCapability] =
+      Set(TableCapability.ACCEPT_ANY_SCHEMA).asJava
+
+    override def vacuum(): Unit = {println("VACUUM!!")}
```

Comment:
1. Where is the usage of this class?
2. Don't use println unless there's clear reason to do so. Use logXXX instead.
3. You may want to add flag here instead of modifying InMemoryTable. Please revert the change of InMemoryTable as it doesn't need to be modified.

Reply:
1. This is a test class.
2/3. replaced this with a flag that is set.

Reply:
If this was replaced, can you remove it?